dht/token.hh

```diff
@@ -218,8 +218,18 @@ inline bool operator<=(const token& t1, const token& t2) { return std::rel_ops::
 inline bool operator>=(const token& t1, const token& t2) { return std::rel_ops::operator>=(t1, t2); }
 std::ostream& operator<<(std::ostream& out, const token& t);
 
+// Returns a successor for token t.
+// Wraps around if there is no successor.
+//
```

Comment:
see:
```
class token {
    // INT64_MIN is not a legal token, but a special value used to represent
    // infinity in token intervals.
    // If a token with value INT64_MIN is generated by the hashing algorithm,
    // the result is coerced into INT64_MAX.
    // (So INT64_MAX is twice as likely as every other token.)
    static inline int64_t normalize(int64_t t) {
        return t == std::numeric_limits<int64_t>::min() ? std::numeric_limits<int64_t>::max() : t;
    }
```

so next_token() call on _data == INT64_MAX will result in INT64_MIN (which is considered an illegal token).

but we don't want to loop around INT64_MAX, if we normalize it like above.

so perhaps we want to just skip INT64_MIN, to continue the iteration from the bottom of the valid token range.

Reply:
Yes. I'll remove the wrap around thing and make it UB instead.